modules/ROOT/pages/dataweave-troubleshoot.adoc

```diff
@@ -129,6 +129,15 @@ Stack Overflow. Max stack is 256
 
 You can configure the maximum stack size by using the property `com.mulesoft.dw.stacksize`.
 
+=== No space left on device
+In order to handle large payloads, DataWeave generates some data which is handled in memory unless it exceeds a certain configurable limit.
+When that happens the data is stored on disk as output, input, and buffer files in a temporary directory. See xref:dataweave-memory-management.adoc[DataWeave Memory Management] for further details.
+
+To avoid the exception `No space left on device` more resource is needed to been able to run the app, or the app needs to be reviewed to reduce its resource consumption.
+
+Normally, once the streams referencing those files are closed the files are released. In case that these types of files remains in the temporal folder even after the flow has completed, this could mean that the streams remain unclosed and cause leaks.
```

Comment:
I would add a note saying that the closure normally happens when flows complete their execution, so long-running and particularly concurrent executions could involve having many of these files living at once. Then the explanation that sometimes bugs can lead to them not being removed and the note for support.